sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/src/GlobalSuppressions.cs

```diff
@@ -0,0 +1,6 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Usage", "AZC0018: Protocol method should have requestContext as the last parameter and don't have model as parameter type or return type. Protocol method should not have optional parameters if ambiguity exists between protocol method and convenience method.", Justification = "Flag keep-non-overloadable-protocol-signature is on.")]
```

Comment:
GlobalSuppressions are atypical. Please use `#pragma` as we typically do inline.

Reply:
There are two reasons:
1. It is generated code. If we add it as method level, then we need to add customization for whole method.
2. We suppress it because it is a GAed library so we need to set `keep-non-overloadable-protocol-signature` flag to not apply new logic. This flag is library level. So every method in this library has to suppress AZC0018.

Reply:
@pshao25 can we have the generator automatically add the suppression when `keep-non-overloadable-protocol-signature` since we know that AZC0018 will occur and the user is asking for us to bypass overloads?

Reply:
@heaths we changed the expected behavior, so there is no global suppression now.